src/scripts/main.js

```diff
@@ -1,5 +1,25 @@
 'use strict';
 
 document.addEventListener('click', e => {
-  // write code here
+  const spider = document.querySelector('.spider');
+  const wall = document.querySelector('.wall');
+  const heightLimit = wall.clientHeight - spider.clientHeight;
+  const widthLimit = wall.clientWidth - spider.clientWidth;
+
+  if (e.target.classList.contains('wall')) {
+    const currentHeight = e.offsetY - spider.clientHeight / 2;
+    const currentWidth = e.offsetX - spider.clientWidth / 2;
+
+    spider.style.top = currentHeight > heightLimit
+      ? heightLimit + 'px'
+      : currentHeight <= 0
+        ? 0 + 'px'
```

Comment:
Don't use nested ternary operators. It's very hard to read.
Try to improve your code (you can split logic and check the "out of boundary" with separate `if` for example)